src/core/log/logger/log.ts

```diff
@@ -0,0 +1,111 @@
+/*!
+ * V4Fire Core
+ * https://github.com/V4Fire/Core
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Core/blob/master/LICENSE
+ */
+
+import $C = require('collection.js');
+import logEngine from 'core/log/engines';
+import * as env from 'core/env';
+import { LogLevel, LogMessageOptions } from 'core/log/types';
+
+interface LogRecord {
+	context: string,
+	logLevel?: LogLevel,
+	details: unknown[],
+}
+
+let
+	options,
+	queue: LogRecord[] = [];
+
+const setConfig = (val) => {
+	options = {
+		patterns: [],
+		...val
+	};
+	options.patterns = $C(options.patterns).map((el) => Object.isRegExp(el) ? el : new RegExp(el));
+};
+
+const
+	defaultContext = 'global';
+
+env.get('log').then(setConfig, setConfig);
+env.event.on('set.log', setConfig);
+env.event.on('remove.log', setConfig);
+
+/**
+ * Call appropriate log engine. Enqueue log records until set options up and then log them.
```

Comment:
Call an appropriate log engine. 
Enqueues log records until set options up and then log them